overlord/snapstate/handlers.go

```diff
@@ -286,6 +288,17 @@ func (m *SnapManager) installOneBaseOrRequired(st *state.State, snapName string,
 		return nil, err
 	}
 	if isInstalled {
+		// check if the default provider can provide the content it's meant to
+		ok, err := hasAllContentAttrs(st, snapName, contentAttrs)
+		if err != nil {
+			return nil, err
+		}
+
+		if !ok {
+			// default provider is missing some content tags (likely outdated) so update it
+			return Update(st, snapName, nil, userID, flags)
```

Comment:
I *think* we can hit store.ErrNoUpdateAvailable here if for whatever reason the content provider hasn't been updated in the store? Is this error going to surface and prevent entire install/update (would be change that breaks compatibility)?

Reply:
Yes, that would prevent the installation. I incorrectly assumed that previously the installation would fail if the plug couldn't be auto-connected but I see that's not true (which makes sense). I can change it to log that snapd failed to update the prereq and proceed with the install